RELEASE/scripts/autoscend/autoscend_migration.ash

```diff
@@ -0,0 +1,142 @@
+script "autoscend_migration.ash"
+
+static string __autoscend_version = "1.0.0";
+static int __autoscend_confirm_timeoutMS = 10000;
+static string __remove_sl_ascend_confirmation = "Looks like you have the old sl_ascend project installed as well. Would you like to remove it? (it is no longer maintained). Will default to false in 10 seconds.";
+static string __migrate_sl_ascend_properties_confirmation = "Looks like you may be migrating from sl_ascend. Starting with a fresh run using autoscend is adviable but we can try to migrate all the sl_ascend properties (results may vary). Will default to true in 10 seconds.";
+static string __migrate_sl_ascend_properties_remove_confirmation = "Would you like to clean up old sl_ascend properties after migrating them? Will default to false in 10 seconds.";
+
+string autoscend_current_version(){
+  if(!property_exists("auto_current_version")){
+    set_property("auto_current_version", __autoscend_version);
+  }
+  return get_property("auto_current_version");
+}
+
+string autoscend_previous_version(){
+  if(!property_exists("auto_prev_version")){
+    set_property("auto_prev_version", "0.0.0");
+  }
+  return get_property("auto_prev_version");
+}
+
+boolean autoscend_needs_update(){
+  if(autoscend_previous_version() == "0.0.0" || autoscend_current_version() != __autoscend_version){
+    if(get_property("auto_need_update").to_boolean()){
+      print("Forcing migration (partially complete migration or user set flag): auto_need_update => true");
+    }
+    set_property("auto_need_update", true);
+  }
+  return get_property("auto_need_update").to_boolean();
+}
+
+boolean autoscend_migrate(){
```

Comment:
should migrate current sl_ascend state (sl_* properties) to the new autoscend properties (auto_* properties) allowing people to switch mid run, in theory. also establishes a pattern of migrating state, if that ever comes up.